lib/player.js

```diff
@@ -703,6 +721,77 @@ shaka.Player = class extends shaka.util.FakeEventTarget {
     if (mediaElement) {
       this.attach(mediaElement, /* initializeMediaSource= */ true);
     }
+
+    if (canvas) {
+      this.canvas_ = canvas;
+    }
+  }
+
+  /**
+   * Create a shaka.lcevc.Dil object
+   * @private
+   */
+  createDIL_() {
+    if (this.lcevcDil_ == null) {
+      this.lcevcDil_ = new shaka.lcevc.Dil(
+          this.video_,
+          this.canvas_,
+          this.lcevcDilConfig_,
+      );
+      this.canvas_ = this.lcevcDil_.canvas_;
```

Comment:
I don't like this circular logic with `canvas_`.  Please let the Player own the reference to the canvas, and push it down to Dil.

Reply:
This is removed as its not needed anymore.